File: Python/알고리즘/SWEA/16183. 전기버스.py
T = int(input())

for tc in range(1, T + 1):
    # K : 버스가 한번에 갈 수 있는 거리
    # N : 시작점과 종점 사이의 거리
    # M : 충전기가 놓여있는 댓수
    K, N, M = map(int, input().split())
    arr = [0] + list(map(int, input().split())) + [N]

    # 충전횟수를 카운트할 변수
    charged = 0
    # 버스의 현재 위치 (출발점부터)
    start = 0
    for i in range(1, M + 2):
        if arr[i] - arr[i - 1] > K:
            charged = 0
            break

        if arr[i] - start > K:
            start = arr[i - 1]
            charged += 1

    print(f'#{tc} {charged}')

File: SWEA/Cpp/16183. 전기버스.cpp
#include <iostream>

using namespace std;

int main()
{
    int T;
    cin >> T;
    for (int i = 0; i < T; i++)
    {
        int K, N, M;
        cin >> K >> N >> M;
        int* arr = new int[M + 2];
        arr[0] = {0};
        arr[M + 1] = {N};
        for (int j = 1; j < M + 1; j++)
            cin >> arr[j];

        int cnt = 0; // 충전 횟수
        int start = 0; // 출발 위치

        for (int x = 1; x < M + 2; x++)
        {
            if (arr[x] - arr[x - 1] > K)
            {
                cnt = 0;
                break;
            }

            if (arr[x] - start > K)
            {
                start = arr[x - 1];
                cnt += 1;
            }
        }
        cout << "#" << i + 1 << " " << cnt << endl;
    }
}